### PR TITLE
Support different templates per function

### LINKED ISSUE #516: Template for session method is not DRY
In both `build/templates/session.py.mako` and `src/nimodinst/templates/session.py.mako` there is a pretty complicated template snippet for rendering a method that's identical in both mako templates.

In `build/templates/session.py.mako`this already lives in its own mako def called `render_method`.

We should eliminate duplication by putting the piece of the template that deals with rendering a method in its own file and use it in both places. I can see it applying to more places in the future, such as NI-TClk which will likely have it's own session template too.

This opens the door to further flexibility, such as allowing us to specify which mako def to use for rendering a method (default to the one we have now). This will allow creating function template definitions for those functions that are different than most other functions in the drivers in ways that don't make sense to add to the generator.

### PR DESCRIPTION
**Pull request by texasaggie97**
[X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
~~[ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
~~[ ] I've added tests applicable for this pull request~~
### What does this Pull Request accomplish?
* Moves session function template definition into separate file
* Add default template for each function that doesn't already have a 'function_template' key

### List issues fixed by this Pull Request below, if any.
* Fix #516 

### What testing has been done?
* Unit
* System
* Visual inspection of diffs

### CODE REVIEW

**Comment by marcoskirsch** [comment]
> Add default template for each function that doesn't already have a 'function_template' key

Technically the template is for a **method** of Session, not a function.

File: build/helper/metadata_add_all.py [hunk 1]
```diff
@@ -149,6 +149,14 @@ def _add_default_value_name_for_docs(parameter, module_name):
 _repeated_capability_parameter_names = ['channelName', 'channelList', 'channel', 'channelNameList']
 
 
+def _add_template_name(f):
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/metadata_add_all.py]: Changed key and function name to 'method_template_filename'

File: build/helper/metadata_add_all.py [hunk 2]
```diff
@@ -149,6 +149,14 @@ def _add_default_value_name_for_docs(parameter, module_name):
...
+    '''Add default function template name if not already defined'''
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/metadata_add_all.py]: Changed

File: build/helper/metadata_add_all.py [hunk 3]
```diff
@@ -149,6 +149,14 @@ def _add_default_value_name_for_docs(parameter, module_name):
...
+    if 'function_template' not in f:
+        f['function_template'] = 'session_default_function.py.mako'
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply build/helper/metadata_add_all.py]: Renamed to 'session_default_method.py.mako'.  I wanted to keep it associated with session.

**Pull request merged**